packages/block-editor/src/components/list-view/use-list-view-scroll-into-view.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+export default function useListViewScrollIntoView( {
+	firstSelectedBlockClientId,
+	numBlocksSelected,
+	selectedItemRef,
+} ) {
+	useEffect( () => {
+		if (
+			numBlocksSelected !== 1 ||
+			! selectedItemRef?.current ||
+			! selectedItemRef.current.scrollIntoView
+		) {
+			return;
+		}
+
+		const { top, height } = selectedItemRef.current.getBoundingClientRect();
+		const { innerHeight } = window;
+
+		// If the selected block is not visible, scroll to it.
+		// The hard-coded value of 110 corresponds to the position at the top of the scrollable area.
```

Comment:
What does the "position at the top of the scrollable area" mean here? Is that the height of the UI at the top?

Reply:
Thanks for the question here — I couldn't quite work out how to describe it. The `top` value is from the top of the screen, but we want to calculate the value in relation to the visible area which is up to the bottom of this tabbed area:

![image](https://user-images.githubusercontent.com/14988353/210908471-94d988bb-1e80-412f-ba61-84a210745ab5.png)

We can get that value by doing a look up using `getScrollContainer` in the `@wordpress/dom` package, but I included a hard-coded value for now, since I think that might be a bit more performant than having to do a look up. Happy to swap it out for the "real" calculation if you think that'd be a better approach (or to try to update the comment to be clearer!)

Reply:
I definitely think it'd be better to use `getScrollContainer`, especially as this UI has undergone recent change, and might end up being changed further.

Reply:
Thanks, I thought there might be a solution that wouldn't cause problems if the UI changed. 👍